core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -167,5 +167,14 @@ private static String pathContext(String tableLocation) {
 
       return resolvedContext;
     }
+
+    private static String computeHash(String fileName) {
+      Preconditions.checkState(fileName != null, "fileName cannot be null");
+      byte[] messageDigest =
+          HASH_FUNC.hashBytes(fileName.getBytes(StandardCharsets.UTF_8)).asBytes();
+      String hash = Base64.getUrlEncoder().encodeToString(messageDigest);
+
+      return hash.substring(0, 8);
```

Comment:
what's the length of the original hash? And is that fixed length or variable?

Reply:
Looking at the javadoc, it always produces 32 bits, and if we base64 encode it to 6 bits per char, it means we will just get 5 chars out of it plus some padding. And from what @danielcweeks suggested before, 5 chars is enough entropy, so we can get the last 30 bits and use that to always encode a 5-char prefix. Is this understanding right?

Reply:
I think it always ends up being 8 characters after the Base64 encoding, but that's only because of the padding.  We need to address the base64 padding characters, but you might be able to just stuff bytes to get there.  Alternatively, we could just drop/replace those characters since they will always come at the end of the hash.

Reply:
Seems like we could just start off with a 6 byte buffer, stuff it with random bytes and then copy the hash bytes to the front.  That would produce a hash without padding:

```java
byte [] hashBuffer = new byte[6];
random.nextBytes(hashBuffer);
byte[] messageDigest = HASH_FUNC.hashBytes(fileName.getBytes(StandardCharsets.UTF_8)).asBytes();
System.arrayCopy(messageDigest, 0, hashBuffer, 0, 4);
String hash = Base64.getUrlEncoder().encodeToString(hashBuffer);
```

Reply:
ACK made the changes

Reply:
@danielcweeks also want to know your thoughts about, if just using pseudo random choosen character from [0-9][ A-Z] [a-z], generating alpha numeric hash is suff, as presently delta just uses this util : 

```
def getRandomPrefix(numChars: Int): String = {
  Random.alphanumeric.take(numChars).mkString
}
```

never the less we are getting a good variance for the first character already with the hash func you suggested already.